src/OpenTelemetry/SuppressInstrumentationScope.cs

```diff
@@ -0,0 +1,74 @@
+﻿// <copyright file="SuppressInstrumentationScope.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using OpenTelemetry.Context;
+
+namespace OpenTelemetry
+{
+    public sealed class SuppressInstrumentationScope : IDisposable
+    {
+        private static readonly RuntimeContextSlot<bool> SuppressInstrumentationRuntimeContextSlot = RuntimeContext.RegisterSlot<bool>("otel.suppress_instrumentation");
+
+        private readonly bool previousValue;
+        private bool disposed;
+
+        internal SuppressInstrumentationScope(bool value = true)
+        {
+            this.previousValue = SuppressInstrumentationRuntimeContextSlot.Get();
+            SuppressInstrumentationRuntimeContextSlot.Set(value);
+        }
+
+        public static implicit operator bool(SuppressInstrumentationScope x) => SuppressInstrumentationRuntimeContextSlot.Get();
```

Comment:
```suggestion
        public static implicit operator bool(SuppressInstrumentationScope unused) => SuppressInstrumentationRuntimeContextSlot.Get();
```

Reply:
Or other name that explains the intention.

Reply:
😄 agree `unused` is better than `x`. I changed it. Though this did leave me wondering if people would prefer to be able to use `_` in this case. I think it's pretty common practice, though I imagine allowing it means allowing any variable to start with a `_` which I'm guessing we don't want. No biggie `unused` is great.

Reply:
I like `_` and initially I was trying to use it, then caught by a style cop 😄